source/includes/dev/_microservice_dev_guide.md

```diff
@@ -1,17 +1,312 @@
+
+- Create a PR from the source branch to the target branch (from `dev` to `staging`, or from `staging` to `production`)
+- Deploy the code to the respective environment (we'll see how in a moment)
+- Test it with actual calls to the API, to validate that no side effects were introduced.
+
+Depending on the scale of the changes you're doing, it's recommended to use [git tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging) with [semantic versioning](https://semver.org/). Also be sure to update the `CHANGELOG.md` accordingly, and the `package.json` or equivalent files if they refer a version number. 
+
+Changes being pushed to either `production` or `staging` should be announced in advance in the `general` channel in the WRI API Slack (and to contact Ethan Roday if you're not in that Slack workspace). Specifically, for changes going to `production`, that notice period should be of at least 14 days, during which said changes should be available in `staging` for testing by everyone. In rare cases, if a hotfix is needed to fix a breaking change in `production`, the 14-day lead time can be circumvented, but an announcement still must be made. 
+
+
+It's also best practice to announce the changes you're about to deploy before doing so, so that other developers of RW API applications can be on the lookout for regressions, and can quickly get in touch with you should any undesired behavior change be detected.
+
+Each of the referred environments lives on a separate [Kubernetes](https://kubernetes.io/) cluster (hosted with [AWS EKS](https://aws.amazon.com/eks/)), and deployment is done using individual Jenkins instances:
+
+- [Jenkins for the dev environment](https://jenkins.aws-dev.resourcewatch.org)
+- [Jenkins for the staging environment](https://jenkins.aws-staging.resourcewatch.org)
+- [Jenkins for the production environment](https://jenkins.aws-prod.resourcewatch.org)
+
+All 3 instances have similar overall configuration, but different microservices may deploy differently depending on the behavior coded into the `Jenkinsfile` that's part of their source code - for example, some WRI sites are also deployed using this approach, but opt to deploy both staging and production versions to the `production` cluster, and may not be in the `staging` or `dev` Jenkins.
+
+The list of jobs you find on each Jenkins instance will match the list of services deployed on that environment. In the details of each job, you should find a branch named after the environment, which corresponds to the Github branch with the same name (some services may still have the old approach, with `develop` for `dev` and `staging`, and `master` for `production`). You may also find other branches, or a different branch structure, depending on the service itself - again, the `Jenkinsfile` configuration is king here, and you should refer to it to better understand what is the desired behavior per branch. In some cases, old branches will be listed on Jenkins but should be ignored.
+
```

Comment:
I understand that there is some variation here due to legacy and projected specific decisions. But I think it is important to highlight again the best practice approach so that we can streamline this more in the future.